tests/test_cliCmdIngestFiles.py

```diff
@@ -0,0 +1,119 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Unit tests for daf_butler CLI ingest-files command.
+"""
+
+import os
+import json
+import unittest
+from astropy.table import Table
+
+from lsst.daf.butler.cli.butler import cli
+from lsst.daf.butler.cli.utils import clickResultMsg, LogCliRunner
+from lsst.daf.butler import Butler
+from lsst.daf.butler.tests import MetricsExample
+from lsst.daf.butler.tests.utils import ButlerTestHelper, makeTestTempDir, MetricTestRepo, removeTestTempDir
+
+
+TESTDIR = os.path.abspath(os.path.dirname(__file__))
+
+
+class CliIngestFilesTest(unittest.TestCase, ButlerTestHelper):
+
+    configFile = os.path.join(TESTDIR, "config/basic/butler.yaml")
+
+    def setUp(self):
+        self.root = makeTestTempDir(TESTDIR)
+        self.addCleanup(removeTestTempDir, self.root)
+
+        self.testRepo = MetricTestRepo(self.root,
+                                       configFile=self.configFile)
+
+        self.root2 = makeTestTempDir(TESTDIR)
+        self.addCleanup(removeTestTempDir, self.root2)
+
+        # Create some test output files to be ingested
+        files = []
+        datasets = []
+        for i in range(2):
+            data = MetricsExample(summary={"int": i, "string": f"{self.id()}_{i}"})
+            outfile = f"test{i}.json"
+            with open(os.path.join(self.root2, outfile), "w") as fd:
+                json.dump(data._asdict(), fd)
+            datasets.append(data)
+            files.append(outfile)
+
+        # Associate the dataId with these datasets
+        self.files = files
+        self.datasets = datasets
```

Comment:
Seems like you could have just defined `self.files` and `self.datasets` right away, at the cost of only a little more verbosity.